### ISSUE TITLE
RFC2307bis 'id' lookups are prohibitively slow

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#803]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/799
- Created at 2011-02-14 19:30:15 by [sgallagh](https://pagure.io/user/sgallagh)
- Closed as Fixed
- Assigned to [jhrozek](https://pagure.io/user/jhrozek)
---

For an LDAP setup using RFC2307bis with many users and nested groups, performing

    id username

can take as long as several minutes to complete.

Jakub did an initial investigation into the code and discovered that we aren't using the {{{sysdb_add_fake_user()}}} routine in the RFC2307bis nested group processing, which means that we're resorting to looking up all of the users in those groups. This is very slow.

### Comments

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2011-02-22 15:11:31](https://pagure.io/SSSD/sssd/issue/799#223759)**

Fields changed

milestone: NEEDS_TRIAGE => SSSD 1.6.0
priority: critical => blocker

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2011-03-02 17:18:07](https://pagure.io/SSSD/sssd/issue/799#223760)**

https://bugzilla.redhat.com/show_bug.cgi?id=681562

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2011-05-04 13:24:16](https://pagure.io/SSSD/sssd/issue/799#223761)**

Fields changed

patch:  => 1
status: new => assigned

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2011-05-20 13:44:22](https://pagure.io/SSSD/sssd/issue/799#223762)**

Fixed in master:

77bc3d93ddd41edee6046508884d7e95553ed5b7

c18bd9de80daf87e33d60c6c44d83b418cfe37b3

6b95a91c1a49c2eff480820cfd8be51d70a29ffe

7bdaf2a712d73763e7c3d25f6bb544b18f7028eb

resolution:  => fixed
status: assigned => closed

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2011-11-21 12:41:07](https://pagure.io/SSSD/sssd/issue/799#223763)**

The problem still exists in 1.6.3

resolution: fixed => 
rhbz:  => 
status: closed => reopened

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2011-11-21 14:13:58](https://pagure.io/SSSD/sssd/issue/799#223764)**

Fields changed

milestone: SSSD 1.6.0 => NEEDS_TRIAGE

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2011-11-21 14:31:09](https://pagure.io/SSSD/sssd/issue/799#223765)**

This request has been coming from several different users running in different environments. I would propose that we gather as much info from the users before we decide on where to optimize. 

I think that a systemtap script that generates a callgraph with statistics about how much time do we spend in different functions and perhaps how much time we spend waiting on I/O (both disk and network) would help us here. Then we would at least know whether we should spend time parallelizing the lookups or perhaps profiling the memberof plugin some more..

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2011-12-17 17:29:21](https://pagure.io/SSSD/sssd/issue/799#223766)**

Replying to [comment:5 mmoeller]:
> The problem still exists in 1.6.3

Marcus, can you measure how slow initgroups are for you?

I would like to ask you to:
1. clear the caches
2. run "id -G $username"

The -G parameter is important - that would make sure only initgroups is performed and "id" would then output the GID numbers of the groups user is a member of. Without the -G parameter, "id" then calls getgrgid() on all the group GIDs which may be very slow (and may not be fixable).

Thank you!

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2011-12-17 17:30:33](https://pagure.io/SSSD/sssd/issue/799#223767)**

Also when running the test, would you mind telling us how big the group structure is? In other words, how many groups does the user belong to?

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2012-01-05 15:44:48](https://pagure.io/SSSD/sssd/issue/799#223768)**

Fields changed

blockedby:  => 
blocking:  => 
milestone: NEEDS_TRIAGE => SSSD Deferred

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2012-01-05 15:45:01](https://pagure.io/SSSD/sssd/issue/799#223769)**

Fields changed

priority: blocker => minor

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2012-01-05 18:54:03](https://pagure.io/SSSD/sssd/issue/799#223770)**

The issue is already fixed in 1.6. The bug was reopened by mistake.

milestone: SSSD Deferred => SSSD 1.6.0
resolution:  => fixed
status: reopened => closed

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2012-01-30 21:22:15](https://pagure.io/SSSD/sssd/issue/799#223771)**

Linked to Bugzilla bug: https://bugzilla.redhat.com/show_bug.cgi?id=742052 (*Red Hat Enterprise Linux 6*)

rhbz:  => [https://bugzilla.redhat.com/show_bug.cgi?id=742052 742052]

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2017-02-24 14:28:36](https://pagure.io/SSSD/sssd/issue/799#223772)**

**Metadata Update from @sgallagh**:
- Issue assigned to jhrozek
- Issue set to the milestone: SSSD 1.6.0
